packages/core/src/utils.mjs

```diff
@@ -94,32 +94,60 @@ export function beamIntersectsY(from, to, y) {
  * @param {Point} a2 - Point 2 of line A
  * @param {Point} b1 - Point 1 of line B
  * @param {Point} b2 - Point 2 of line B
- * @return {Point} intersections - The Point at the intersection
+ * @return {Point|false} intersections - The Point at the intersection or `false` if the lines are parallel
  */
 export function beamsIntersect(a1, a2, b1, b2) {
-  let slopeA = a1.slope(a2)
-  let slopeB = b1.slope(b2)
-  if (slopeA === slopeB) return false // Parallel lines
-
-  // Check for vertical line A
-  if (Math.round(a1.x * 10000) === Math.round(a2.x * 10000))
-    return new Point(a1.x, slopeB * a1.x + (b1.y - slopeB * b1.x))
-  // Check for vertical line B
-  else if (Math.round(b1.x * 10000) === Math.round(b2.x * 10000))
-    return new Point(b1.x, slopeA * b1.x + (a1.y - slopeA * a1.x))
-  else {
-    // Swap points if line A or B goes from right to left
-    if (a1.x > a2.x) a1 = a2.copy()
-    if (b1.x > b2.x) b1 = b2.copy()
-    // Find y intercept
-    let iA = a1.y - slopeA * a1.x
-    let iB = b1.y - slopeB * b1.x
-
-    // Find intersection
-    let x = (iB - iA) / (slopeA - slopeB)
-    let y = slopeA * x + iA
-
-    return new Point(x, y)
+  const intersection = beamIntersection(a1, a2, b1, b2)
+  if (!intersection) return false
+  return intersection.p
+}
+
+/**
+ * Finds the intersection of two endless lines (beams)
+ *
+ * @param {Point} a1 - Point 1 of line A
+ * @param {Point} a2 - Point 2 of line A
+ * @param {Point} b1 - Point 1 of line B
+ * @param {Point} b2 - Point 2 of line B
+ * @return {{p:Point, t: number, u:number}|false} the intersection.
+ * The method will return `false` if the lines are (approximately) parallel or undefined,
+ * e.g., if both points of a line have (approximately) the same coordinate.
+ * Otherwise, p is the point of the intersection,
+ * t and u determine where the intersection lies relative to the points of line A and line B respectively.
+ * 0.0 means the intersection is on the first point, 1.0 on the second point.
+ */
+function beamIntersection(a1, a2, b1, b2) {
```

Comment:
note: Not sure if we want to export this method. Access to t and u values could be useful for some designs, but would add something that requires stable versioning,